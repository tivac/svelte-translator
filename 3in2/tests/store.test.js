"use strict";

const { writable } = require("svelte3/store");
const Adapter = require("../store.js");

const {
    readableSimple,
    writableSimple,
    readableObject,
    writableObject
} = require("./specimens/store.js");

describe("3in2 store adapter", () => {
    let root;

    beforeEach(() => {
        root = document.createElement("div");
    });

    it("should render readable stores that return a simple value", async () => {
        const Component = require("./specimens/readable-simple.html");
        
        new Component({
            target : root
        });
        
        expect(root.innerHTML).toMatchSnapshot();
        
        readableSimple._store._write(1);
        
        expect(root.innerHTML).toMatchSnapshot();
    });
            
    it("should render readable stores that return an object", async () => {
        const Component = require("./specimens/readable-object.html");

        new Component({
            target : root
        });

        expect(root.innerHTML).toMatchSnapshot();

        readableObject._store._write({ fooga : "fooga", wooga : "wooga" });

        expect(root.innerHTML).toMatchSnapshot();
    });
    
    it("should render writable stores that return a simple value", async () => {
        const Component = require("./specimens/writable-simple.html");
        
        new Component({
            target : root
        });
        
        expect(root.innerHTML).toMatchSnapshot();
        
        writableSimple._store.set(1);
        
        expect(root.innerHTML).toMatchSnapshot();
    });
            
    it("should render writable stores that return an object", async () => {
        const Component = require("./specimens/writable-object.html");

        new Component({
            target : root
        });

        expect(root.innerHTML).toMatchSnapshot();

        writableObject._store.set({ fooga : "fooga", wooga : "wooga" });

        expect(root.innerHTML).toMatchSnapshot();
    });

    it("should update the underlying store when .set() is called", async () => {
        const original = writable(0);
        const store = new Adapter(original);

        store.set(1);

        expect(store.get()).toMatchSnapshot();
        
        const unsubscribe = original.subscribe((value) => expect(value).toMatchSnapshot());

        unsubscribe();
    });

    it("should throw if .set() is called against a readable store", async () => {
        expect(() => readableSimple.set(0)).toThrowErrorMatchingSnapshot();
    });
});
