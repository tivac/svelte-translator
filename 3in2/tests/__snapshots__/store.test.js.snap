// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3in2 store adapter should render readable stores that return a simple value 1`] = `

<div>
  0
</div>

`;

exports[`3in2 store adapter should render readable stores that return a simple value 2`] = `

<div>
  1
</div>

`;

exports[`3in2 store adapter should render readable stores that return an object 1`] = `

<div>
  0 - 0
</div>

`;

exports[`3in2 store adapter should render readable stores that return an object 2`] = `

<div>
  fooga - wooga
</div>

`;

exports[`3in2 store adapter should render writable stores that return a simple value 1`] = `

<div>
  0
</div>

`;

exports[`3in2 store adapter should render writable stores that return a simple value 2`] = `

<div>
  1
</div>

`;

exports[`3in2 store adapter should render writable stores that return an object 1`] = `

<div>
  0 - 0
</div>

`;

exports[`3in2 store adapter should render writable stores that return an object 2`] = `

<div>
  fooga - wooga
</div>

`;

exports[`3in2 store adapter should throw if .set() is called against a readable store 1`] = `"Called .set() on a readable store"`;

exports[`3in2 store adapter should update the underlying store when .set() is called 1`] = `
Object {
  "value": 1,
}
`;

exports[`3in2 store adapter should update the underlying store when .set() is called 2`] = `1`;
